### PR TITLE
Added Houser Robber DP psuedocode

### PR DESCRIPTION
**Pull request by bicycleman15**

### CODE REVIEW

**Comment by ghost** [comment]
First of all, the name of the file should be `SourceCode.tex`. Please rename it.

**Comment by ghost** [comment]
You should've consulted me before writing the pseudocode. Next time, please discuss the approach and the plan for writing the pseudocode

File: Pseudocode/Dynamic Programming/House Robber/houserobber.tex [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+\begin{algorithm}
+
+  \caption{Find the maximum amount of money that can be robbed from non-adjacent houses}
+  
+  \begin{algorithmic}[1]
+    \Ensure Zero Based Indexing for the array
+    \Statex
+    
+    \Function{$rob$}{$arr$}
```
Comment (ghost) [inline Pseudocode/Dynamic Programming/House Robber/houserobber.tex]: Name the function `House_Robber`

File: Pseudocode/Dynamic Programming/House Robber/houserobber.tex [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+\captionsetup[algorithm]{labelformat = algnonumber}
+
+\begin{document}
+
+\begin{algorithm}
+
+  \caption{Find the maximum amount of money that can be robbed from non-adjacent houses}
+  
+  \begin{algorithmic}[1]
+    \Ensure Zero Based Indexing for the array
```
Comment (ghost) [inline Pseudocode/Dynamic Programming/House Robber/houserobber.tex]: Use 1 based indexing throughout your code.

File: Pseudocode/Dynamic Programming/House Robber/houserobber.tex [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    \Statex
+    
+    \Function{$rob$}{$arr$}
+    
+        \LeftComment{$dp[i]$ represents the maximum money we can rob using first $i$ houses}
+        
+        \Statex
+        \State $n \gets arr.length$
+        
+        \Statex
```
Comment (ghost) [inline Pseudocode/Dynamic Programming/House Robber/houserobber.tex]: A useless new line. The code looks quite separated. Remove it.

File: Pseudocode/Dynamic Programming/House Robber/houserobber.tex [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        
+        \State $dp[0] \gets arr[0]$
+        \State $dp[1] \gets max(arr[0],arr[1])$
+        
+        \Statex
```
Comment (ghost) [inline Pseudocode/Dynamic Programming/House Robber/houserobber.tex]: We also don't need this line. Remove it

File: Pseudocode/Dynamic Programming/House Robber/houserobber.tex [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        
+        \For {$i = 2$ \textbf{to} $n-1$}
+            \State $dp[i] = max(dp[i-1], arr[i] + dp[i-2])$
+        \EndFor
+        
+        \Statex
+        
+        \State \Return $dp[n-1]$
```
Comment (ghost) [inline Pseudocode/Dynamic Programming/House Robber/houserobber.tex]: Update accordingly

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
Please make the required changes. (I'll suggest more once you're done with this).

File: Pseudocode/Dynamic Programming/House Robber/houserobber.tex [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    \Function{$rob$}{$arr$}
+    
+        \LeftComment{$dp[i]$ represents the maximum money we can rob using first $i$ houses}
+        
+        \Statex
+        \State $n \gets arr.length$
+        
+        \Statex
+        
+        \State $dp[0] \gets arr[0]$
```
Comment (ghost) [inline Pseudocode/Dynamic Programming/House Robber/houserobber.tex]: Don't assume anything about the inputs. What if negative values are allowed? Handle it by updating to `max(arr[0], 0)`

File: Pseudocode/Dynamic Programming/House Robber/houserobber.tex [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        \State $dp[1] \gets max(arr[0],arr[1])$
+        
+        \Statex
+        
+        \For {$i = 2$ \textbf{to} $n-1$}
+            \State $dp[i] = max(dp[i-1], arr[i] + dp[i-2])$
```
Comment (ghost) [inline Pseudocode/Dynamic Programming/House Robber/houserobber.tex]: For clarity, break it into parts such as 

`\State $take\_it \gets arr[i] + dp[i-2]$`
`\State $leave\_it \gets dp[i-1]$`
`\State $dp[i] \gets max(take\_it, \ leave\_it)$`

**Comment by ghost** [comment]
Also, your code doesn't handle the case when the array is of length 1. Handle it explicitly

**Comment by bicycleman15** [comment]
I have done the required changes. Please have a look.

File: Pseudocode/Dynamic Programming/House Robber/source.tex [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+        \State $n \gets arr.length$
+        
+        
+        \State $dp[1] \gets max(0,arr[1])$
+        \Statex
+        \If{$n$ is 1}
+        \State \Return $dp[1]$
+        \EndIf
+        \Statex
+        \State $dp[2] \gets max(arr[1],arr[2])$
```
Comment (ghost) [inline Pseudocode/Dynamic Programming/House Robber/source.tex]: Change it to `\State $dp[2] \gets max(arr[1], \ arr[2], \ 0)$`.

Note: This is to handle the case when all the elements are negative.

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
Looks good. I have just requested a minor change. Correct it and click on `Resolve Conversation` for each comment. Ping me once you are done.

**Comment by ghost** [comment]
Also, I see that you haven't changed the file name. Rename it to `SourceCode.tex` (Notice the capitalization).

**Comment by bicycleman15** [comment]
Okay I have done the changes.

**Comment by ghost** [comment]
Good work! Thank you for your contribution to this repository. I hope you got to learn something new from this experience.

**Comment by ghost** [comment]
Closes #46

**Pull request merged**